supervisor/main.go

```diff
@@ -0,0 +1,183 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"flag"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/LeonidVas/tarantool-supervisor/supervisor/api/supervisorhttp"
+	"github.com/LeonidVas/tarantool-supervisor/supervisor/core"
+)
+
+// args describes the parsed arguments.
+type args struct {
+	//CfgPath - path to Supervisor config.
+	CfgPath string
+	// Addr - address to start the HTTP server(host:port).
+	Addr string
+}
+
+// parseArgs returns the parsed arguments.
+func parseArgs() *args {
+	var args args
+	flag.StringVar(&args.CfgPath, "cfg", "cfg.json",
+		"path to Supervisor config.")
+	flag.StringVar(&args.Addr, "addr", "127.0.0.1:8080",
+		"address to start the HTTP server(host:port).")
+	flag.Parse()
+
+	return &args
+}
+
+// parseCfg parse the Supervisor JSON config.
+func parseCfg(path string) (*core.Cfg, error) {
+	// Check is the file exists.
+	if _, err := os.Stat(path); err != nil {
+		return nil, err
+	}
+
+	// Open the file.
+	jsonFile, err := os.Open(path)
+	if err != nil {
+		return nil, err
+	}
+	defer jsonFile.Close()
+
+	// Set defaults.
+	cfg := core.Cfg{
+		InstancesDir: "/etc/tarantool/supervisor/instances",
+		TermTimeout:  30,
+	}
+
+	// Read and parse config.
+	byteValue, err := ioutil.ReadAll(jsonFile)
+	if err != nil {
+		return nil, err
+	}
+	if err = json.Unmarshal(byteValue, &cfg); err != nil {
+		return nil, err
+	}
+
+	// In the config, the time is indicated in seconds. Convert the value.
+	cfg.TermTimeout = cfg.TermTimeout * time.Second
+
+	return &cfg, nil
+}
+
+// handleZombie handles zombie child processes, if any, in a non-blocking style.
+func handleZombie(sv *core.Supervisor) {
+	// Get PID of the terminated Instance.
+	pid, err := syscall.Wait4(-1, nil, syscall.WNOHANG, nil)
+
+	// If the error "no child processes" (syscall.Errno(10)) or pid == 0 -
+	// this means that the process was stopped as planned, and the wait ()
+	// function has already been called.
+	if pid == 0 || (err != nil && errors.Is(err, syscall.Errno(10))) {
+		return
+	} else if err != nil {
+		log.Printf(`Can't get the PID of a process by using "Wait4". Error: "%v"`,
+			err)
+		return
+	}
+
+	// If the Restartable Instance flag is true, try to restart the Instance.
+	if id, err := sv.RestartAfterTermInstance(pid); err != nil {
+		log.Printf(`Can't restart the Instance. Old PID : %v. ID: %v. Error: "%v"`,
+			pid, id, err)
+	} else {
+		log.Printf("The Instance has been restarted. ID: %v", id)
```

Comment:
Maybe it's better to return it as error and then print in function that call this?

Reply:
BTW, why not `log.Warn`?

Reply:
> Maybe it's better to return it as error and then print in function that call this?

I think it's okay to print the log at this level.

Reply:
> BTW, why not `log.Warn`?

Because `log.Warn` doesn't exist.

Reply:
Cartridge CLI uses https://github.com/apex/log. It supports different log levels and formats.

Reply:
> Cartridge CLI uses https://github.com/apex/log. It supports different log levels and formats.

I suggest adding a more functional logger later, at this stage (MVP) the capabilities of the standard one are quite enough.